### DISCUSSION TITLE
Toggle dock icon on macos

### DISCUSSION THREAD

**Discussion by arichyx** [tauri-apps/tauri#10774] [Q&A]
I'm developing a macos application. When the app starts, I hope it only show in menu bar, doesn't show in dock. And there is a `settings` menu item in the menu, a window would appear after clicking the `settings` menu item. When the window appears, I want the app icon to show in dock. When the window gets closed, I want the app icon to hide in dock.

I've tried `app.set_activation_policy(tauri::ActivationPolicy::Accessory));` to hide it when app starts, but I cannot access `app` in `on_system_tray_event` and `on_window_event`. I need to show app icon in dock when the window appears.

**Comment by ollyde** [comment]
Also face the same issue, did you find a solution?

**Reply by qutek** [reply]
i was able to do that with something like this

```
// auto-hide dock icon on macOS
if let Some(main_win) = app.get_webview_window("main") {
    let app_handle = app.handle().clone();
    let main_for_events = main_win.clone();
    main_win.on_window_event(move |event| {
        #[cfg(target_os = "macos")]
        if let WindowEvent::CloseRequested { api, .. } = event {
            api.prevent_close();
            let _ = main_for_events.hide();
            let _ = app_handle.set_activation_policy(ActivationPolicy::Accessory);
        }
    });
}
```

**Reply by ollyde** [reply]
One works on launch no?

**Reply by qutek** [reply]
yeah, on launch / show the window, you should set it back to `ActivationPolicy::Regular`
```
#[cfg(target_os = "macos")]
{
    let _ = app.set_activation_policy(ActivationPolicy::Regular);
}
```
